service/worker/parentclosepolicy/workflow.go

```diff
@@ -160,19 +168,81 @@ func ProcessorActivity(ctx context.Context, request Request) error {
 			})
 		}
 
+		switch typedErr := err.(type) {
+		case nil:
+			processor.metricsClient.IncCounter(metrics.ParentClosePolicyProcessorScope, metrics.ParentClosePolicyProcessorSuccess)
+		case *serviceerror.NotFound:
+			// no-op
+		case *serviceerror.NamespaceNotActive:
+			remoteExecutions[typedErr.ActiveCluster] = append(remoteExecutions[typedErr.ActiveCluster], execution)
+		default:
+			processor.metricsClient.IncCounter(metrics.ParentClosePolicyProcessorScope, metrics.ParentClosePolicyProcessorFailures)
+			getActivityLogger(ctx).Error("failed to process parent close policy", tag.Error(err))
+			return err
+		}
+	}
+
+	if err := signalRemoteCluster(
+		ctx,
+		processor.clientBean,
+		request.ParentExecution,
+		remoteExecutions,
+		processor.cfg.NumParentClosePolicySystemWorkflows(),
+	); err != nil {
+		getActivityLogger(ctx).Error("Failed to signal remote parent close policy workflow", tag.Error(err))
+		return err
+	}
+
+	return nil
+}
+
+func signalRemoteCluster(
+	ctx context.Context,
+	clientBean client.Bean,
+	parentExecution commonpb.WorkflowExecution,
+	remoteExecutions map[string][]RequestDetail,
+	numWorkflows int,
+) error {
+	for cluster, executions := range remoteExecutions {
+		remoteClient := clientBean.GetRemoteFrontendClient(cluster)
+
+		signalValue := Request{
+			ParentExecution: parentExecution,
+			Executions:      executions,
+		}
+		signalInput, err := converter.GetDefaultDataConverter().ToPayloads(signalValue)
 		if err != nil {
-			if _, ok := err.(*serviceerror.NotFound); ok {
-				err = nil
-			}
+			return err
 		}
 
+		signalCtx, cancel := context.WithTimeout(ctx, signalTimeout)
+		_, err = remoteClient.SignalWithStartWorkflowExecution(
+			signalCtx,
+			&workflowservice.SignalWithStartWorkflowExecutionRequest{
+				Namespace:  common.SystemLocalNamespace,
+				RequestId:  uuid.New(),
+				WorkflowId: getWorkflowID(numWorkflows),
+				WorkflowType: &commonpb.WorkflowType{
+					Name: processorWFTypeName,
+				},
+				TaskQueue: &taskqueue.TaskQueue{
+					Name: processorTaskQueueName,
+				},
+				Input:                 nil,
+				WorkflowTaskTimeout:   timestamp.DurationPtr(workflowTaskTimeout),
+				Identity:              common.WorkerServiceName + "-service",
```

Comment:
what about add cluster name in the identity?

Reply:
Good idea 👍 . Let me check how to get current cluster name here.

Reply:
If it is going to touch many places, I am fine without the cluster name. I just feel it may help with debugging.

Reply:
it's a simply change. I have added current cluster name to processor's params and used it in the identity.